Add highlight and show how to use some features

minervapanda opened the pull request:
![ezgif com-video-to-gif](https://cloud.githubusercontent.com/assets/11734962/19299453/00e9d3ca-9072-11e6-8068-8df8595f7673.gif)
Solves : Highlight and show how to use some features of the app which are not easily visible - #161

smarx commented:
*Automated message from Dropbox CLA bot*

@minervapanda, it looks like you've already signed the Dropbox CLA. Thanks!

minervapanda commented:
I could not upload the whole video.
This pr addresses : 
[x] Swipe to remove the chatbox
[x] Switching streams from the menu
[x] Private chat drawer
[x] Long press on message for more options

kunall17 commented:
Thankyou for working on this! 

- Rather than creating activities which shows static images, you could easily make multiple showcase views like it is done here https://github.com/amlcurran/ShowcaseView/blob/master/sample/src/main/java/com/github/amlcurran/showcaseview/sample/MemoryManagementTesting.java (Got this one from this issue https://github.com/amlcurran/ShowcaseView/issues/325) in the ZulipActivity.java file
<br> And while switching showcases you could programitically open/close the drawers etc 
- Add a help option in the overFlow menu which would show these highlights
- Tweak with background and foreground text Color as the current scheme makes it hard to read the text. Maybe decrease the opacity of the background if possible!

kunall17 commented:
And also remove the commit 	a475f45 from this PR, as this is not related to this feature!

minervapanda commented:
Hi @kunall17 , I added the help option to the menu and similar way of implementation is in ChatBoxFeature java file. Give a look. We need to show different layouts for different demonstration for other features.
Also removed the unwanted commits

brainwane commented:
@minervapanda could you add a little more detail to the commit message (customs explained https://zulip.readthedocs.io/en/latest/version-control.html#commit-messages ) to list the features that get demonstrated in this new showcase?

brainwane commented on app/src/main/java/com/zulip/android/activities/ShowFeature.java:
```diff
@@ -0,0 +1,54 @@
+        showcaseView = new ShowcaseView.Builder(this)
+                .withNewStyleShowcase()
+                .setTarget(new ViewTarget(R.id.results, this))
+                .setContentTitle("Swipe to the Left")
```
I think we don't capitalize Left here.

brainwane commented on app/src/main/java/com/zulip/android/activities/ChatBoxFeature.java:
```diff
@@ -0,0 +1,63 @@
+                    showcaseView = new ShowcaseView.Builder(ChatBoxFeature.this)
+                            .withNewStyleShowcase()
+                            .setTarget(new ViewTarget(R.id.switch_stream, ChatBoxFeature.this)).setContentTitlePaint(titlePaint)
+                            .setContentTitle("Press here to choose a stream")
```
I think we usually use "Tap" here instead of "Press".

niftynei commented:
Hey @minervapanda, have you had a chance to look over the suggestions from Sumana and Kunal yet?

yadav-rahul commented:
@minervapanda Due to recent merges in `master`, merge conflicts have been created. Please try to resolve them.

saketkumar95 commented:
@yadav-rahul  This PR is no longer valid. 
@timabbott @kunall17  I think this should be closed.
